### PR TITLE
show detail of user

### PR DESCRIPTION
**Pull request by ghost**
## Description
Display username, email, join_date, role ( member or admin), books were borrowed, books donated,
number of ratings, name of books is borrowing
## Reference Requirements
- Product Backlog ID: PBI-12
- Task ID: 2.3, 10.1, 10.2, 10.3, 12.1, 12.2

## Migrations
YES 

## Impacted Areas in Application
List general components of the application that this PR will affect:

- app/Model/User.php
- resources/lang/en/dashboard.php

## SQL query and `explain`
- List of all queries and attach the result of each explaining query (in text, screenshot...)
 + Get all infomation of user and total_borrowed, total_donated
SELECT users.*,
COUNT(DISTINCT(borrowings.id)) AS total_borrowed, 
COUNT(DISTINCT(donators.id)) AS total_donated
FROM users LEFT JOIN borrowings 
ON borrowings.user_id = users.id 
LEFT JOIN donators ON donators.user_id = users.id
GROUP BY users.id

 + Get book is being borrowed by user
EXPLAIN select `books`.`name` 
from `borrowings` 
inner join `books` on `borrowings`.`book_id` = `books`.`id` 
inner join `users` on `borrowings`.`user_id` = `users`.`id` 
where `users`.`employee_Code` = 'AT-00002' and `borrowings`.`to_date` is null 
limit 1

- Refer to [Using Explain with MySQL]
image
image

### CODE REVIEW

**Comment by at-luanle** [comment]
LGTM

File: phpunit.dusk.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (at-huydq) [inline phpunit.dusk.xml]: file này sao cũng push lên vậy. lần sau nhớ kiểm tra kỹ đã push nhé.

File: app/Http/Controllers/Admin/UserController.php [hunk 2]
```diff
@@ -35,10 +35,33 @@ public function index()
...
+
+        $user = User::select($fields)
+        ->leftJoin('borrowings', 'users.id', '=', 'borrowings.user_id')
+        ->leftJoin('donators', 'users.id', '=', 'donators.user_id')
+        ->where('users.employee_Code', '=', $employeeCode)
+        ->groupBy('users.id')
+        ->first();
+
+        $bookBorrowing = DB::table('borrowings')
+        ->Join('books', 'borrowings.book_id', '=', 'books.id')
```
Comment (at-huydq) [inline app/Http/Controllers/Admin/UserController.php]: join viết thường nhé, nhớ tuân thủ convention.

**Review by ghost — COMMENTED** [review]
please check my comment

File: phpunit.dusk.xml (hunk 1 above)
Comment (at-nghiavo) [reply phpunit.dusk.xml]: em đã fix, a check lại hộ em

**Comment by at-huydq** [comment]
LGTM.

**Pull request merged**